src/core/namespace.c

```diff
@@ -1339,6 +1361,37 @@ static int mount_private_sysfs(const MountEntry *m, const NamespaceParameters *p
         return mount_private_apivfs("sysfs", mount_entry_path(m), "/sys", /* opts = */ NULL, p->runtime_scope);
 }
 
+static bool check_recursiveprot_supported(void) {
+        int r;
+
+        /* memory_recursiveprot is only supported for kernels >= 5.7. */
```

Comment:
Maybe, better to extend the comment more, that `mount_option_supported()` uses fsopen() and fsconfig(), which are added in 5.2. So, if this is failed, we can assume that the option is not supported.